node/cn/handler_test.go

```diff
@@ -58,3 +63,173 @@ func TestNewProtocolManager(t *testing.T) {
 		assert.Equal(t, errIncompatibleConfig, err)
 	}
 }
+
+func TestSampleSize(t *testing.T) {
+	peers := make([]Peer, minNumPeersToSendBlock-1)
+	assert.Equal(t, len(peers), sampleSize(peers))
+
+	peers = make([]Peer, 4)
+	assert.Equal(t, minNumPeersToSendBlock, sampleSize(peers))
+
+	peers = make([]Peer, 16)
+	assert.Equal(t, 4, sampleSize(peers))
+}
+
+func TestSamplePeersToSendBlock(t *testing.T) {
+	pm := &ProtocolManager{}
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	block := newBlock(blockNum1)
+	hash := block.Hash()
+
+	// 1. When the given node is a consensus node.
```

Comment:
For all test case, why don't you input same mocked `CNWithoutBlock`/`PNWithoutBlock`/`PeersWithoutBlockExceptCN`/`ENWithoutBlock`?

Because depend on the network topology, `samplePeersToSendBlock` can be at various situation.

Reply:
@KimKyungup Right. This is the thing that @nogang told me yesterday. That way is the correct way to test. However, this test will be added to `peer_set_test.go` as it does not seem related to `ProtocolManager`. If given nodeType, all things can be done inside the `PeerSet` interface. I will update the PR.